File: Fortran/3.5/sgemv_openacc.f90
subroutine gemv_parallel(n, A, x, y)
    implicit none
    integer, intent(in) :: n
    real, intent(in) :: A(n, n)
    real, intent(in) :: x(n)
    real, intent(out) :: y(n)
    integer :: i, j
    real :: sum

    !$ACC PARALLEL LOOP
    do i = 1, n
        sum = 0.0
        do j = 1, n
            sum = sum + A(i, j) * x(j)
        end do
        y(i) = sum
    end do
    !$ACC END PARALLEL LOOP

end subroutine gemv_parallel

File: C/3.5-prompt/Par/sgemv_openacc.c
void sgemv(int m, int n, float alpha, float *a, float *x, float beta, float *y)
{
    #pragma acc parallel loop collapse(2) independent
    for (int i = 0; i < m; i++)
    {
        float sum = 0.0;
        for (int j = 0; j < n; j++)
        {
            sum += a[i * n + j] * x[j];
        }
        y[i] = alpha * sum + beta * y[i];
    }
}
